Add tailscale-dev/deck-tailscale dir to PATH

gustakasn0v opened the pull request:
As discussed in https://github.com/tailscale-dev/deck-tailscale/pull/15 and https://discord.com/channels/960281551428522045/1186652921186766898/1186663913878736937 we're changing the Tailscale setup on Steam Deck to install it in `/opt/tailscale`. One of the side-effects of this is the `tailscale` CLI not being available under `/usr` anymore, and Decky plugins can't rely on $PATH because Decky Loader doesn't start a login session (`/etc/profile`, `/etc/bash.bashrc` etc aren't available)

This PR adds the new path to the plugin's Python backend. It does it in a backwards-compatible way, so it'll work even if users don't upgrade their `tailscale-dev/deck-tailscale` install. Plus it adds a few minor local development quality-of-life improvements

Commit:
Add tailscale-dev/deck-tailscale dir to PATH

Commit:
Ignore deck.json, pnpm cache

saumya-banthia commented:
@gustakasn0v That's great, does the change you mention from the commit make it so that tailscale installs persist across updates (without having to rerun the install command once every update)?

Also, @japzone1 this could be a solution to the problem you showcased in #7 of not having the specific directory in path, if you have not already tried it out. I could not get tailscale working with Nix, otherwise I would have tried myself (if you were previously using the tailscale plugin with the official method, it would be of help if you could share how you switched over to using Nix; I did already go through the reddit post it did not go into detail on removing previous tailscale installs and other guides lead to **tailscaled** getting broken)

japzone1 commented:
@saumya-banthia Ooo, if you can update PATH in the plugin like that, then adding the below as well should fix Nix installs as well: 

    os.environ['PATH'] += ':/home/deck/.nix-profile/bin'

> I could not get tailscale working with Nix, otherwise I would have tried myself (if you were previously using the tailscale plugin with the official method, it would be of help if you could share how you switched over to using Nix; I did already go through the reddit post it did not go into detail on removing previous tailscale installs and other guides lead to tailscaled getting broken)

Ah, my Nix guide assumed a fresh SteamOS that hadn't used the previous method, since I developed it on my new OLED model. If you wanted to switch over you'd have to make sure the Systemd extension was disabled/removed so that there isn't extra Tailscale binaries in PATH, and the `tailscaled.service` file in `/etc/systemd/system/` was modified according to my guide so that it points to the proper `tailscaled` from Nix.

gustakasn0v commented:
> @gustakasn0v That's great, does the change you mention from the commit make it so that tailscale installs persist across updates (without having to rerun the install command once every update)?

Yes, that's exactly the goal. You can read more about the options we considered in https://github.com/tailscale-dev/deck-tailscale/issues/13 and https://github.com/tailscale-dev/deck-tailscale/pull/15

> Also, @japzone1 this could be a solution to the problem you showcased in #7 of not having the specific directory in path

> if you can update PATH in the plugin like that, then adding the below as well should fix Nix installs as well: `os.environ['PATH'] += ':/home/deck/.nix-profile/bin'`

Yes, this would fix Nix installs as well, @saumya-banthia lmk if you want this plugin to support multiple install types, and I can add that line to this PR. Personally if we fix the deck-tailscale script to persist across updates I don't see the need for a Nix install, but I'm happy either way

saumya-banthia commented on .vscode/config.sh:
could you remove this commit from the PR, it is generally safer to have executable scripts uploaded with 644 permissions, instead of 755 for public repos.

gustakasn0v replied:
Gotcha, will do

saumya-banthia commented:
> I can add that line to this PR

Please do. Also add below change to the file to ensure non-duplicate entries in the path (except ones that already exist)

EDIT: replaced for loop with a comprehension

```
user_dirs = ['/opt/tailscale', '/home/deck/.nix-profile/bin']

current_path = os.environ["PATH"].split(":")

new_path = ":".join(current_path + [user_dir for user_dir in user_dirs if user_dir not in current_path])
os.environ["PATH"] = new_path
```

> if we fix the deck-tailscale script to persist across updates I don't see the need for a Nix install

I too do not, maybe we deprecate use of nix installs in the future but if people are using Nix right now and it is not taking up **much additional effort to maintain** (_in this case us needing to add additional dir is part of keeping up with the official method, which so happens to be the case with nix install as well_), I am okay with supporting it.

Commit:
Add Nix path, prevent duplicate paths

gustakasn0v commented:
@saumya-banthia I've addressed all comments, ready for another review

edit: P.S. that python oneliner was lit, I learned so much just by looking at it, thanks!

saumya-banthia commented:
Merging and closing

gustakasn0v commented:
Thanks! 

Do you know what the process is to get this published to the Decky store? Happy to wait until the device list is ready and then get both features published.

saumya-banthia commented:
Hey @gustakasn0v, I made an oopsie making a hard-reset to a pre-merge commit, which seems to have erased the commits you made, could you raise another PR, with the same commits now originating at the new main branch of this repo.

As for the publishing, it'll be done once I raise a PR to the Decky plugin db repo: https://github.com/SteamDeckHomebrew/decky-plugin-database (which would be as soon as I merge the new PR you make)

EDIT: To make it simpler you could just copy the file changes from the relevant files from this PR (there is a tab called "Files Changed" at the top of this PR). I apologize for causing a double spend in effort due to a mistake on my part. This is the reason why a new PR would be needed: https://stackoverflow.com/questions/12674304/github-reopening-a-merged-pull-request

gustakasn0v commented:
No worries, it's an easy fix! I'll do this later today.

Please share the Decky store PR link here when you have it, I'm 1. curious to see how the process is 2. excited to test this all out through the store instead of my local dev builds!

saumya-banthia commented:
> No worries, it's an easy fix! I'll do this later today.
> 
> Please share the Decky store PR link here when you have it, I'm 1. curious to see how the process is 2. excited to test this all out through the store instead of my local dev builds!

For sure, I will share the PR request link. Meanwhile, if you want to have a look at the structure, here's another one of my plugins' update PR: https://github.com/SteamDeckHomebrew/decky-plugin-database/pull/417

> edit: P.S. that python oneliner was lit, I learned so much just by looking at it, thanks!

Just saw this. Thanks. list and dict comprehensions are huge help at cleaning up code 😉
